Src/Base/AMReX_OpenMP.cpp

```diff
@@ -0,0 +1,175 @@
+#include <AMReX_OpenMP.H>
+#include <AMReX.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_Print.H>
+
+#if defined(__APPLE__)
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#endif
+
+#if defined(_WIN32)
+#include <windows.h>
+#endif
+
+#include <fstream>
+#include <iostream>
+#include <optional>
+#include <set>
+#include <sstream>
+#include <string>
+#include <thread>
+#include <vector>
+
+
+namespace amrex::OpenMP
+{
+    int
+    numUniquePhysicalCores ()
```

Comment:
This function is in `amrex::OpenMP::` here, but the declaration is in `amrex::` in AMReX_OpenMP.H. So let's move this to `amrex::`.